### PR TITLE
[Component with its own class name as service type] LPS-196947 Refactor components from echo modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-196947

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
@MrJohn1911 tests didn't run due to:

```
     [exec] > Task :apps:application-list:application-list-user-personal-site-permissions:compileJava FAILED
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:13: error: package com.liferay.osgi.service.tracker.collections does not exist
     [exec] import com.liferay.osgi.service.tracker.collections.EagerServiceTrackerCustomizer;
     [exec]                                                    ^
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:14: error: package com.liferay.osgi.service.tracker.collections.map does not exist
     [exec] import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
     [exec]                                                        ^
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:15: error: package com.liferay.osgi.service.tracker.collections.map does not exist
     [exec] import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
     [exec]                                                        ^
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:118: error: cannot find symbol
     [exec] 	private ServiceTrackerMap<String, PanelApp> _serviceTrackerMap;
     [exec] 	        ^
     [exec]   symbol:   class ServiceTrackerMap
     [exec]   location: class com.liferay.application.list.user.personal.site.permissions.internal.model.listener.CompanyModelListener
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:268: error: cannot find symbol
     [exec] 			implements EagerServiceTrackerCustomizer<PanelApp, PanelApp> {
     [exec] 			           ^
     [exec]   symbol:   class EagerServiceTrackerCustomizer
     [exec]   location: class com.liferay.application.list.user.personal.site.permissions.internal.model.listener.CompanyModelListener.UserPersonalSitePermissions
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:88: error: cannot find symbol
     [exec] 		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
     [exec] 		                     ^
     [exec]   symbol:   variable ServiceTrackerMapFactory
     [exec]   location: class com.liferay.application.list.user.personal.site.permissions.internal.model.listener.CompanyModelListener
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:276: error: method does not override or implement a method from a supertype
     [exec] 			@Override
     [exec] 			^
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:313: error: method does not override or implement a method from a supertype
     [exec] 			@Override
     [exec] 			^
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:323: error: method does not override or implement a method from a supertype
     [exec] 			@Override
     [exec] 			^
     [exec] 9 errors
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-09/test-1-28/test-portal-acceptance-pullrequest(master)/7744/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
@MrJohn1911 startup error:

```
[beanshell] 2023-09-28 00:16:07.601 ERROR [SystemExecutorServiceUtil-8][ROOT:47] bundle com.liferay.application.list.user.personal.site.permissions:6.0.13 (1372)[com.liferay.application.list.user.personal.site.permissions.internal.model.listener.CompanyModelListener(6331)] : The activate method has thrown an exception
[beanshell] org.osgi.framework.InvalidSyntaxException: Missing attr at "(&(objectClass=com.liferay.application.list.PanelApp)(panel.category.key=site_administration*))=*)(objectClass=com.liferay.application.list.PanelApp))": (&((&(objectClass=com.liferay.application.list.PanelApp)(panel.category.key=site_administration*))=*)(objectClass=com.liferay.application.list.PanelApp))
[beanshell] 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse_attr(FilterImpl.java:1666) ~[org.eclipse.osgi.jar:?]
[beanshell] 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse_item(FilterImpl.java:1595) ~[org.eclipse.osgi.jar:?]
[beanshell] 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse_filtercomp(FilterImpl.java:1539) ~[org.eclipse.osgi.jar:?]
[beanshell] 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse_filter(FilterImpl.java:1505) ~[org.eclipse.osgi.jar:?]
[beanshell] 	at org.eclipse.osgi.internal.framework.FilterImpl$Parser.parse_and(FilterImpl.java:1554) ~[org.eclipse.osgi.jar:?]
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-09/test-1-35/test-portal-acceptance-pullrequest(master)/4629/functional-smoke-tomcat90-postgresql144-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
ci:test:echo

**Comment by Kvale1733** [comment]
@MrJohn1911 No regressions found.

File: modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java [hunk 1]
```diff
@@ -57,6 +76,32 @@ public void onAfterCreate(Company company) {
 			});
 	}
 
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_userPersonalSitePermissions = new UserPersonalSitePermissions();
+
+		String filter = StringBundler.concat(
```
Comment (shuyangzhou) [inline modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:83]: This is not right, with this change you will end up with filter having duplicated content.
```"(&(&(objectClass=com.liferay.application.list.PanelApp)(panel.category.key=site_administration*))(objectClass=com.liferay.application.list.PanelApp))"```

Also don't do inner class in another inner class, that's just unnecessary structure complexity.

File: modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java [hunk 2]
```diff
@@ -67,6 +112,228 @@ public void onAfterCreate(Company company) {
...
+					throw throwable;
+				}
+			}
+
+			@Override
+			public void modifiedService(
+				ServiceReference<PanelApp> serviceReference,
+				PanelApp panelApp) {
+
+				removedService(serviceReference, panelApp);
```
Comment (shuyangzhou) [inline modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java:320]: You don't consume any property from the ServiceReference, there is nothing to modify here.

**Review by shuyangzhou — COMMENTED** [review]
See the inline comments.
Please redo and resend just the content from the 1st commit.
Cleanup the service tracker collection usage within the UserPersonalSitePermissions first, then merge it into CompanyModelListener.

Resend the rest of the pull after the 1st commit is re-done properly.

**Pull request closed without merging**